deeplearning4j-zoo/src/main/java/org/deeplearning4j/zoo/model/AlexNet.java

```diff
@@ -38,26 +40,17 @@
  * Weight distribution uses 0.1 std for all layers in the paper but 0.005 in the dense layers in the imagenetExample code
  *
  */
-@NoArgsConstructor
+@AllArgsConstructor
+@Builder
 public class AlexNet extends ZooModel {
 
-    private int[] inputShape = new int[] {3, 224, 224};
-    private int numLabels = 1000;
-    private long seed = 42;
-    private WorkspaceMode workspaceMode;
-    private ConvolutionLayer.AlgoMode cudnnAlgoMode;
-
-    public AlexNet(int numLabels, long seed) {
-        this(numLabels, seed, WorkspaceMode.ENABLED);
-    }
-
-    public AlexNet(int numLabels, long seed, WorkspaceMode workspaceMode) {
-        this.numLabels = numLabels;
-        this.seed = seed;
-        this.workspaceMode = workspaceMode;
-        this.cudnnAlgoMode = workspaceMode == WorkspaceMode.ENABLED ? ConvolutionLayer.AlgoMode.PREFER_FASTEST
-                        : ConvolutionLayer.AlgoMode.NO_WORKSPACE;
-    }
+    @Builder.Default private long seed = 1234;
+    @Builder.Default private int[] inputShape = new int[] {3, 224, 224};
+    private int numLabels;
+    @Builder.Default private IUpdater updater = new Nesterovs(1e-2, 0.9);
+    @Builder.Default private CacheMode cacheMode = CacheMode.DEVICE;
+    @Builder.Default @Deprecated private WorkspaceMode workspaceMode = WorkspaceMode.ENABLED;
```

Comment:
We may want to keep this non-deprecated, for benchmarks and the like?

Reply:
Agreed.